urllib3/connectionpool.py

```diff
@@ -790,11 +790,13 @@ def _prepare_proxy(self, conn):
             set_tunnel = conn.set_tunnel
         except AttributeError:  # Platform-specific: Python 2.6
             set_tunnel = conn._set_tunnel
-
+        host = self.host
+        if not (host.startswith('[') and host.endswith(']')):
+            host = '[' + host + ']'
```

Comment:
This is definitely wildly overbroad. For example, if we're tunneling to `example.com`, this will turn the host into `[example.com]` (definitely incorrect).

Do you mind making this more subtle, so that we have some idea of what the actual host is? Probably this will need to be done at a higher level where we have a complete URL object in hand.